md/moves.py

```diff
@@ -56,3 +68,197 @@ def n_accepted(self):
     @property
     def n_proposed(self):
         return np.sum(self.n_proposed_by_move)
+
+
+class NPTMove(MonteCarloMove):
+    def __init__(
+        self,
+        ubps,
+        lamb,
+        masses,
+        temperature,
+        pressure,
+        n_steps,
+        dt=1.5e-3,
+        friction=1.0,
+        barostat_interval=5,
+        seed=None,
+    ):
+
+        intg = lib.LangevinIntegrator(temperature, dt, friction, masses, seed)
+        self.integrator_impl = intg.impl()
+        all_impls = [bp.bound_impl(np.float32) for bp in ubps]
+
+        bond_list = get_bond_list(ubps[0])
+        group_idxs = get_group_indices(bond_list)
+
+        barostat = lib.MonteCarloBarostat(len(masses), pressure, temperature, group_idxs, barostat_interval, seed + 1)
+        barostat_impl = barostat.impl(all_impls)
+
+        self.bound_impls = all_impls
+        self.barostat_impl = barostat_impl
+        self.lamb = lamb
+        self.n_steps = n_steps
+
+    def move(self, x: CoordsVelBox):
+        # note: context creation overhead here is actually very small!
+        ctxt = custom_ops.Context(
+            x.coords, x.velocities, x.box, self.integrator_impl, self.bound_impls, self.barostat_impl
+        )
+
+        # arguments: lambda_schedule, du_dl_interval, x_interval
+        _ = ctxt.multiple_steps(self.lamb * np.ones(self.n_steps), 0, 0)
+        x_t = ctxt.get_x_t()
+        v_t = ctxt.get_v_t()
+        box = ctxt.get_box()
+
+        after_npt = CoordsVelBox(x_t, v_t, box)
+
+        self.n_proposed += 1
+        self.n_accepted += 1
+
+        return after_npt
+
+
+class DeterministicMTMMove:
+    def move(self, xvb: CoordsVelBox) -> CoordsVelBox:
+        self.n_proposed += 1
+        y_proposed, acceptance_probability, sub_key = self.acceptance_probability(xvb.coords, xvb.box, self.rng_key)
+        self.rng_key = sub_key
+        alpha = np.random.rand()
```

Comment:
Should this line also use / update `self.rng_key`? (`acceptance_probability` will be a deterministic function of `self.rng_key` state, but the outcome of the "biased coin toss" will depend on global numpy rng state)